cmake-ide.el

```diff
@@ -279,11 +280,12 @@ the closest possible matches available in cppcheck."
 
 (defun cmake-ide--comp-db-file-name ()
   "The name of the compilation database file."
-  (expand-file-name "compile_commands.json" (cmake-ide--get-build-dir)))
+  (when (cmake-ide--get-build-dir)
+    (expand-file-name "compile_commands.json" (cmake-ide--get-build-dir))))
 
 (defun cmake-ide--need-to-run-cmake ()
   "If CMake needs to be run or not."
-  (and (not (get-process "cmake")) ; don't run if already running
+  (and (cmake-ide--get-project-key) (not (get-process "cmake")) ; don't run if already running
```

Comment:
?

Reply:
The logic should be to run CMake if the current buffer is visiting a file that's part of a CMake project. So maybe `cmake-ide--locate-cmakelists`?

Reply:
Maybe, but as far as I understand `cmake-ide-locate-cmakelists` do not actually check if a cmakelists.txt file exists if `cmake-ide--project-dir-var` is set. This might be a bug although (maybe I introduced ... not sure)